Add axis support for `tf.nn.crelu`

yongtang opened the pull request:
This fix tries to address the issue raised in #15619 where it was not possible to specify an `axis` for
`tf.nn.crelu`. By default, `axis=-1` was used for concatenation implicitly.

This fix adds the support of `axis` for `tf.nn.crelu`, and adds test cases for it.

This fix fixes #15619.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang committed:
Add axis support for tf.nn.crelu

This fix tries to address the issue raised in 15619
where it was not possible to specify an `axis` for
`tf.nn.crelu`. By default, `axis=-1` was used for concatenation
implicitly.

This fix adds the support of `axis` for tf.nn.crelu,
and adds test cases for it.

This fix fixes 15619.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang committed:
Adds test cases for axis support with `tf.nn.crelu`

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

caisq commented on tensorflow/python/kernel_tests/relu_op_test.py:
```diff
@@ -441,6 +441,30 @@ def testNumbers(self):
+                           [9, 0, 5, 0, 1],
+                           [0, 3, 0, 7, 0]])
+      self.assertAllEqual(np_crelu, tf_relu)
+      self.assertShapeEqual(np_crelu, crelu)
```
This is not necessary, because `assertAllEqual` checks equality of shapes.
Same below.

yongtang replied:
Thanks @caisq. Unnecessary `assertAllEqual` have been removed.

caisq commented:
Add @martinwicke for API change review.

yongtang committed:
Address review feedback and remove unneeded `assertShapeEqual`,

as it has been covered by `assertAllEqual`

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang commented:
@caisq Thanks for the review. The PR has been updated.

yongtang commented:
By the way, there are quite some failures on Jenkins due to:
```
docker: Error response from daemon: devmapper: Thin Pool has 981366 free data blocks which is less than minimum required 983040 free data blocks. Create more free space in thin pool or use dm.min_free_space option to change behavior.
```

I don't know enough about the infrastructure platform of TensorFlow for Jenkins servers, though I would suggest to take a look at overlay2 storage driver (vs.  devmapper). 

The overlay2 is considered as mature as of now. On most of the Linux platforms it probably would be performant than devmapper. (There might be issues with REHL 7's SELinux but that may have been fixed as well.)

If the docker version on Jenkins is newer enough then overlay2 probably will be an option to consider.

asimshankar commented:
Looks good for API change.

caisq commented:
@tensorflow-jenkins test this please

yongtang committed:
Update golden API with

```
bazel-bin/tensorflow/tools/api/tests/api_compatibility_test
           --update_goldens True
```

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang commented:
Thanks @asimshankar @caisq for the review. The Jenkins failed because of the Api changes. As API review already done, I updated golden with
```
bazel-bin/tensorflow/tools/api/tests/api_compatibility_test
           --update_goldens True
```

and pushed the PR.

The Jenkins should pass now.

caisq commented:
@tensorflow-jenkins test this please
